Simplenote/BackgroundRefreshManager.swift

```diff
@@ -0,0 +1,101 @@
+import Foundation
+import BackgroundTasks
+
+class BackgroundRefreshManager: NSObject {
+    private var timer: Timer? {
+        didSet {
+            oldValue?.invalidate()
+        }
+    }
+
+    private var handler: (()->Void)?
+
+    var finished: Bool = false {
+        didSet {
+            if finished == true {
+                handler?()
+            }
+        }
+    }
```

Comment:
I don't think we need to track `finished`

Reply:
A totally.  The finished variable was something I had put in because I was having issues making an instance of the backgroundFetchManager in the app delegate.  Was trying to use changing it to trigger ending the refresh, but using the handler instead can serve that same purpose.  I have removed the finished var.